common/src/state/ui.rs

```diff
@@ -52,6 +61,51 @@ impl EmojiCounter {
             .map(|(emoji, usage)| (emoji.clone(), *usage))
             .collect()
     }
+
+    pub fn get_matching_emoji(&self, pattern: &str, exact: bool) -> Vec<(String, String)> {
+        if pattern.is_empty() {
+            return vec![];
+        }
+        let mut matches: HashMap<String, String> = default_emoji_list()
+            .iter()
+            .filter_map(|(emoji, alias)| {
+                if (exact && (*alias).eq(pattern)) || (!exact && alias.starts_with(pattern)) {
+                    Some((emoji.to_string(), alias.to_string()))
+                } else {
+                    None
+                }
+            })
+            .collect();
+        for (_, matcher) in self.emoji_filters.iter() {
+            matcher(pattern, exact).iter().for_each(|(emoji, alias)| {
+                matches.insert(emoji.clone(), alias.clone());
+            });
+        }
+        let mut matches: Vec<(String, String)> = matches
+            .iter()
+            .map(|(emoji, alias)| (emoji.clone(), alias.clone()))
+            .collect();
+        matches.sort_by(|(emoji, _), (emoji2, _)| {
+            let first = self.list.get(emoji).unwrap_or(&0_u64);
```

Comment:
the name `list` confused me. I see that `EmojiCounter`'s `list` is really a `HashMap` of integers. This name is bad.